backend/generating/SupervisionAgenda.java

```diff
@@ -0,0 +1,41 @@
+package generation;
+
+import java.util.List;
+
+public class SupervisionAgenda {
+
+    private List<Task> taskList;
+
+    public SupervisionAgenda(List<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    public List<Task> getTaskList(){ return taskList; }
+
+    public StringBuilder writeTasks(List<Task> taskList){
+        StringBuilder toPrint = new StringBuilder();
+        for (Task task : taskList){
+            //formatting non-exam questions as a default; exam questions need to be formatted in the UI
+            if (!task.isSubTask()){
+                // will print out in the form of \section{insert task here}
+                toPrint.append("\\section{" + task.getTaskText() + "}" + "\n");

```

Comment:
If we use a LaTeX` \question` environment (similar to the `\examquestion` environment currently) instead of `\section` then I think we could simplify the Java code a lot. All \question has to do is:

1. `\item`
2. Print out question text in bold
3. Insert either automarkable environment or some comment for the student saying "please insert your answer here)
4. if it has subtasks, do what you're currently doing

I think we want as little distinction between a question and subquestion in the Java code as possible.

and then we enclose the whole document in one big `\begin{enumerate}` (can be hard-coded in the LaTeX).